src/main/java/com/ontotext/kafka/GraphDBSinkTask.java

```diff
@@ -26,32 +21,46 @@
  */
 public class GraphDBSinkTask extends SinkTask {
 
-	private static final Logger LOG = LoggerFactory.getLogger(GraphDBSinkTask.class);
-	private Repository repository;
-	private final AtomicBoolean shouldRun = new AtomicBoolean(true);
-	private final ConcurrentLinkedQueue<Collection<SinkRecord>> sinkRecords = new ConcurrentLinkedQueue<>();
+	private static final Logger log = LoggerFactory.getLogger(GraphDBSinkTask.class);
+
+	/**
+	 * Use a single processor and queue for all tasks of a single unique connector.
+	 */
+	private static final Map<String, SinkRecordsProcessor> processors = new HashMap<>();
 	private GraphDBSinkConfig config;
-	private SinkRecordsProcessor recordProcessor;
-	private Thread recordProcessorThread;
+	private SinkRecordsProcessor processor;
 
 
 	@Override
-	public String version() {
-		return VersionUtil.getVersion();
+	public void start(Map<String, String> properties) {
+		this.config = new GraphDBSinkConfig(properties);
+		log.info("Starting the GraphDB sink task for connector {}", config.getConnectorName());
+		this.processor = startProcessor(config);
+		log.info("Configuration complete.");
 	}
 
-	@Override
-	public void start(Map<String, String> properties) {
-		LOG.info("Starting the GraphDB sink task");
-		config = new GraphDBSinkConfig(properties);
-		repository = initializeRepository(repository, config);
-		LOG.info("Initialized GraphDB repository connection: repository {}, GraphDB instance {}", config.getRepositoryId(),
-			config.getServerUrl());
-		recordProcessor = createProcessor();
-		recordProcessorThread = new Thread(recordProcessor);
-		shouldRun.set(true);
-		recordProcessorThread.start();
-		LOG.info("Configuration complete.");
+	/**
+	 * Creates a new {@link SinkRecordsProcessor} and starts its thread, for a corresponding connector name.
+	 * If a context has already been created (i.e. from another task initialization for the same connector), return the existing queue, without starting
+	 * any new threads.
+	 *
+	 * @param config - The configuration for which to create the new context
+	 * @return ProcessorContext
+	 */
+	private SinkRecordsProcessor startProcessor(GraphDBSinkConfig config) {
```

Comment:
Maybe a better name for this method would be createProcessor because the SinkExecutor class also has a startNewProcessor method

Reply:
While `startprocessor` is not sufficient, neither is `createProcessor` (both miss the fact that this method both creates and starts a new processor) . I think a good alternative would be `createAndStartProcessor`